data_safe_haven/administration/users/user_handler.py

```diff
@@ -81,98 +74,88 @@ def get_usernames(self) -> Dict[str, List[str]]:
             usernames[f"SRE {sre_name}"] = self.get_usernames_guacamole(sre_name)
         return usernames
 
-    def get_usernames_azure_ad(self) -> List[str]:
+    def get_usernames_azure_ad(self) -> list[str]:
         """Load usernames from Azure AD"""
         return [user.username for user in self.azure_ad_users.list()]
 
-    def get_usernames_domain_controller(self) -> List[str]:
+    def get_usernames_domain_controller(self) -> list[str]:
         """Load usernames from all domain controller"""
         return [user.username for user in self.active_directory_users.list()]
 
-    def get_usernames_guacamole(self, sre_name: str) -> List[str]:
+    def get_usernames_guacamole(self, sre_name: str) -> list[str]:
         """Load usernames from Guacamole"""
         return [user.username for user in self.sre_guacamole_users[sre_name].list()]
 
-    def list(self) -> None:
+    def list(self) -> None:  # noqa: A003
         """List Active Directory, AzureAD and Guacamole users
 
         Raises:
-            DataSafeHavenUserHandlingException if the users could not be listed
+            DataSafeHavenUserHandlingError if the users could not be listed
         """
         try:
             # Load usernames
             usernames = self.get_usernames()
             # Fill user information as a table
-            user_headers = ["username"] + list(usernames.keys())
+            user_headers = ["username", *list(usernames.keys())]
             user_data = []
             for username in sorted(set(sum(usernames.values(), []))):
                 user_memberships = [username]
                 for category in user_headers[1:]:
-                    user_memberships.append(
-                        "x" if username in usernames[category] else ""
-                    )
+                    user_memberships.append("x" if username in usernames[category] else "")
                 user_data.append(user_memberships)
 
             # Write user information as a table
             for line in self.logger.tabulate(user_headers, user_data):
                 self.logger.info(line)
         except Exception as exc:
-            raise DataSafeHavenUserHandlingException(
-                f"Could not list users.\n{str(exc)}"
-            ) from exc
+            msg = f"Could not list users.\n{exc!s}"
```

Comment:
This is actually new to me. So, I kind of dislike it because it looks unusual. However, it is shorter and part of the f-string grammar.

The rule is [this one](https://beta.ruff.rs/docs/rules/explicit-f-string-type-conversion/).

I think all of these are unnecessary though as the default is to use `str()`. So I think we should remove all of these.

Reply:
Fixed in ef21e9a6